src/compositedataframe.jl

```diff
@@ -1,23 +1,93 @@
 using Compat
 
-export AbstractCompositeDataFrame, CompositeDataFrame
+export AbstractCompositeDataFrame, AbstractCompositeDataFrameRow, 
+       CompositeDataFrame, row
 
+"""
+    AbstractCompositeDataFrame
+    
+An abstract type that is an AbstractDataFrame. Each type that inherits from
+this is expected to be a type-stable data frame. 
+"""
 abstract AbstractCompositeDataFrame <: AbstractDataFrame
 
+abstract AbstractCompositeDataFrameRow
+
+
+"""
+    row(cdf::AbstractCompositeDataFrame, i)
+    
+Return the `CompositeDataFrameRow` object for row `i` of `cdf`. This object has
+the same fields as `cdf` where the type of each field is taken from the `eltype`
+of the field in `cdf`. 
+
+See also `eachrow(cdf)`.
+
+```julia
+df = CompositeDataFrame(x = 1:3, y = [2, 1, 6])
+dfr = row(df, 3)
+dfr.y   # 6
+```
+"""
+row() = nothing
+
+"""
+```julia
+CompositeDataFrame(columns::Vector{Any}, cnames::Vector{Symbol})
+CompositeDataFrame(columns::Vector{Any}, cnames::Vector{Symbol}, typename::Symbol)
+CompositeDataFrame(; kwargs...)
+CompositeDataFrame(typename::Symbol; kwargs...)
+```
+
+A constructor of AbstractCompositeDataFrames that mimics the `DataFrame` 
+constructor. 
+
+This uses `eval` to create a new type within the current module. 
+
+### Arguments
+
+* `columns` : contains the contents of the columns
+* `cnames` : the names of the columns
+* `typename` : the optional name of the type created
+* `kwargs` : the key gives the column names, and the value is the column contents
+
+### Returns
```

Comment:
In the recent guidelines, we decided that the return value would generally better be documented directly in the function description